src/datahike/db.cljc

```diff
@@ -137,6 +137,68 @@
       (update :aevt di/-persistent!)
       (update :avet di/-persistent!)))
 
+(defn contextual-search-fn [{:keys [temporal?]}]
+  (case temporal?
+    true dbs/temporal-search
+    false dbs/search-current-indices))
+
+(defn contextual-search [db pattern context]
+  ((contextual-search-fn context) db pattern))
+
+(defn contextual-datoms [db index-type cs {:keys [temporal?]}]
+  (case temporal?
+    true (dbu/temporal-datoms db index-type cs)
+    false (di/-slice
+           (get db index-type)
+           (dbu/components->pattern db index-type cs e0 tx0)
+           (dbu/components->pattern db index-type cs emax txmax)
+           index-type)))
+
+(defn contextual-seek-datoms [db index-type cs {:keys [temporal?]}]
+  (case temporal?
+    true (dbs/temporal-seek-datoms db index-type cs)
+    false (di/-slice (get db index-type)
+                     (dbu/components->pattern db index-type cs e0 tx0)
+                     (datom emax nil nil txmax)
+                     index-type)))
+
+(defn contextual-rseek-datoms [db index-type cs {:keys [temporal?]}]
+  (case temporal?
+    true (dbs/temporal-rseek-datoms db index-type cs)
+    false (-> (di/-slice (get db index-type)
+                         (dbu/components->pattern db index-type cs e0 tx0)
+                         (datom emax nil nil txmax)
+                         index-type)
+              vec
+              rseq)))
+
+(defn contextual-index-range [db avet attr start end {:keys [temporal? current-db]}]
+  {:pre [(or (not temporal?) current-db)]}
+  (case temporal?
+    true (dbs/temporal-index-range db current-db attr start end)
+    false (do (when-not (dbu/indexing? db attr)
+                (raise "Attribute"
+                       attr "should be marked as :db/index true" {}))
+              
+              (dbu/validate-attr
+               attr (list '-index-range 'db attr start end) db)
+              (di/-slice
+               avet
+               (dbu/resolve-datom db nil attr start nil e0 tx0)
+               (dbu/resolve-datom db nil attr end nil emax txmax)
+               :avet))))
+
+(defn deeper-index-range [origin-db db attr start end context]
+  (dbi/-index-range origin-db
+                    attr
+                    start
+                    end
+                    (merge {:current-db db}
+                           context)))
+
+(defmacro todo [x] x)
```

Comment:
Is this a leftover?

Reply:
Yes, thanks for spotting that. I have removed it.